### DISCUSSION TITLE
How to access the control instance from the view model

### DISCUSSION THREAD

**Discussion by zpccode** [canton7/Stylet#349] [Q&A] [answered]
How to access the control instance from the view model
E.g:
1.Save the picture of the control screen
2.Canvas dynamically add lines

**Comment by canton7** [comment, marked as answer]
The general approach is not to do this: MVVM strong discourages interacting directly with the view from the VM. If you really need to, you can access the `View` property on the ViewModel, and cast it to your view type.

How exactly you go about it depends on what problem you're solving. For dynamically adding lines to a canvas for example, you would normally have the VM expose a `BindableCollection` containing a list of objects describing your lines (e.g. start and end point). Then you might be able to bind to that in your XAML, or you might need to bind it to an attached behaviour or read it in the codebehind.

**Reply by zpccode** [reply]
Thank you very much, I have a requirement to save the image of the control to a local folder.
My current solution:
1. Create an Event in the viewmodel
2. Get DataContext(viewmodel) in View.xaml.cs, and create an event handler method
3. Through the viewmodel invoked Event.

this work, is it safe to do so?

**Reply by canton7** [reply]
That's one way of doing it. If you'd prefer to use bindings, you can use something like an `IObservable<T>`.
